packages/kit/src/runtime/client/client.js

```diff
@@ -1814,7 +1819,30 @@ export function goto(url, opts = {}) {
 		);
 	}
 
-	return _goto(url, opts, 0);
+	/**
+	 * @type {{
+	 *   replaceState?: boolean;
+	 *   noScroll?: boolean;
+	 *   keepFocus?: boolean;
+	 *   invalidateAll?: boolean;
+	 *   invalidated?: Array<((url: URL) => boolean)>;
+	 *   state?: Record<string, any>
+	 * }}
+	 */
+	const options = opts;
```

Comment:
I think this just stores the object reference and we're still mutating the same object later on.